ci/build/release-prep.sh

```diff
@@ -0,0 +1,117 @@
+    echo "We use this with the release-github-draft.sh and release-github-assets.sh scripts."
+    echo -e "See docs here: https://github.com/github/hub#installation"
+    exit
+  fi
+
+  # Check that they have jq installed
+  if ! command -v jq &>/dev/null; then
+    echo "jq could not be found."
+    echo "We use this to parse the package.json and grab the current version of code-server."
+    echo -e "See docs here: https://stedolan.github.io/jq/download/"
+    exit
+  fi
+
+  # Check that they have rg installed
+  if ! command -v rg &>/dev/null; then
+    echo "rg could not be found."
+    echo "We use this when updating files across the codebase."
+    echo -e "See docs here: https://github.com/BurntSushi/ripgrep#installation"
+    exit
+  fi
+
+  # Check that they have sd installed
+  if ! command -v sd &>/dev/null; then
+    echo "sd could not be found."
+    echo "We use this when updating files across the codebase."
+    echo -e "See docs here: https://github.com/chmln/sd#installation"
+    exit
+  fi
+
+  # Check that they have node installed
+  if ! command -v node &>/dev/null; then
+    echo "node could not be found."
+    echo "That's surprising..."
+    echo "We use it in this script for getting the package.json version"
+    echo -e "See docs here: https://nodejs.org/en/download/"
+    exit
+  fi
+
+  # credit to jakwuh for this solution
+  # https://gist.github.com/DarrenN/8c6a5b969481725a4413#gistcomment-1971123
+  CODE_SERVER_CURRENT_VERSION=$(node -pe "require('./package.json').version")
+  # Ask which version we should update to
+  # In the future, we'll automate this and determine the latest version automatically
+  echo "Current version: ${CODE_SERVER_CURRENT_VERSION}"
+  # The $'\n' adds a line break. See: https://stackoverflow.com/a/39581815/3015595
+  read -r -p "What version of code-server do you want to update to?"$'\n' CODE_SERVER_VERSION_TO_UPDATE
+
+  echo -e "Great! We'll prep a PR for updating to $CODE_SERVER_VERSION_TO_UPDATE\n"
+  # I can't tell you why but
+  # when searching with rg, the version needs to in this format: '3\.7\.5'
+  # that's why we have the parameter expansion with the regex
+  $CMD rg -g '!yarn.lock' -g '!*.svg' --files-with-matches "${CODE_SERVER_CURRENT_VERSION//\./\\.}" | $CMD xargs sd "$CODE_SERVER_CURRENT_VERSION" "$CODE_SERVER_VERSION_TO_UPDATE"
+
+  # Ensure the tests are passing and code coverage is up-to-date
+  echo -e "Running unit tests and updating code coverage...\n"
+  $CMD yarn test:unit
+  # Updates the Lines badge in the README
+  $CMD yarn badges
+  # Updates the svg to be green for the badge
+  $CMD sd "red.svg" "green.svg" ../../README.md
+
+  $CMD git add . && $CMD git commit -am "chore(release): bump version to $CODE_SERVER_VERSION_TO_UPDATE"
+
+  CURRENT_BRANCH=$(git branch --show-current)
+  # Note: we need to set upstream as well or the gh pr create step will fail
+  # See: https://github.com/cli/cli/issues/575
+  $CMD git push -u origin "$CURRENT_BRANCH"
```

Comment:
This could be sketchy, most people probably use `origin` as the name for upstream but it's no guarantee.

Also my version of git doesn't seem to recognize `--show-current`? (I'm on 2.17.1.)

Maybe we can do a check if upstream is set and error if it isn't using something like:

```
git rev-parse --abbrev-ref --symbolic-full-name @{u}
```

Reply:
Or maybe it's sufficient to just `git push`?

Reply:
> Also my version of git doesn't seem to recognize --show-current? (I'm on 2.17.1.)

Time to update 😛 jk, looks like it's new as of [2.22](Also my version of git doesn't seem to recognize --show-current? (I'm on 2.17.1.)).

I should use something more backwards-compatible. 

> This could be sketchy, most people probably use origin as the name for upstream but it's no guarantee.

Good call! 

> git rev-parse --abbrev-ref --symbolic-full-name @{u}

🤯 time to go look up what this does.

> git push -u ${u}

oo that could work too. Either way, this is great feedback. I'll refactor.

Reply:
Had to go with something different. I'll push up and would be curious to get your feedback